### DISCUSSION TITLE
When using "filter" event in Custom API Hook, how do I obtain the whole record instead of just the altered fields?

### DISCUSSION THREAD

**Discussion by powerseed** [directus/directus#12622] [Archive]
The goal is to make one field required if **another** field is filled with certain values, otherwise it is optional. 

For example, field `A` is required when field `B` is true, otherwise field `A` is optional.

I am looking at "filter" event in Custom API Hook (https://docs.directus.io/extensions/hooks/#filter).

The doc says that
```
The callback function itself receives three parameters:

The modifiable payload
An event-specific meta object
A context object
```

However, none of these 3 parameters give me the value of field `B` of the current record.

For example, if I want to update field `A` only, then the first parameter, `The modifiable payload`, will only contains the value of field `A`, but as mentioned above I need the value of field `B` to determine if field `A` is required and if the update operation should be allowed.

Could anyone teach me how to achieve the goal?

Thanks!

**Comment by rijkvanzanten** [comment]
You'd have to read the item you're looking for from the database directly. You should be able to use the ItemsService as follows:

```js
// hook boilerplate

const { ItemsService } = services;

const myCollection = new ItemsService('my_collection', { schema, accountability });

const item = await myCollection.readOne(15, { fields: ['*'] });
```

**Reply by powerseed** [reply]
Could you tell me what kind of values should be passed in to `schema` and `accountability`?

I didn't find a doc about this.

The `readOne` method throws an exception of `Cannot read property 'myCollection' of undefined`, I think is because the `schema` is not fetched correctly. 

Thanks

**Reply by rijkvanzanten** [reply]
Those are passed in the context object of the filter handler function: https://docs.directus.io/extensions/hooks/#filter

So you get something like

```js
function registerHooks({ filter }) {
  filter("items.create", async (input, { schema, accountability }) => {
    const myCollection = new ItemsService('my_collection', { schema, accountability });
    // etc
  });
}
```

**Reply by powerseed** [reply]
Got it. Thank you so much!
